Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -222,6 +237,50 @@ public class PushDeploymentController : ApiController
             repository.Commit(zipDeploymentInfo.Message, zipDeploymentInfo.Author, zipDeploymentInfo.AuthorEmail);
         }
 
+        private async Task LocalZipFetchAndDeploy(IRepository repository, DeploymentInfoBase deploymentInfo, ILogger logger, ITracer tracer)
+        {
+            var zipDeploymentInfo = (ZipDeploymentInfo)deploymentInfo;
+
+            // For this kind of deployment, RepositoryUrl is a local path to the temp zip.
+            var sourceZipFile = zipDeploymentInfo.RepositoryUrl;
+            // the name of the deployed file in D:\home\data\SitePackages\{name}.zip is the 
+            // timestamp in the format yyyMMddHHmmss. 
+            var fileName = $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}.zip";
+            var filePath = Path.Combine(_environment.SitePackagesPath, fileName);
+
+            FileSystemHelpers.EnsureDirectory(_environment.SitePackagesPath);
+
+            var info = FileSystemHelpers.FileInfoFromFileName(sourceZipFile);
+            var sizeInMb = (info.Length / (1024f * 1024f)).ToString("0.00", CultureInfo.InvariantCulture);
+
+            var message = String.Format(
+                CultureInfo.InvariantCulture,
+                "Deploying {0} to {1} with size of {2}",
+                info.FullName,
+                filePath,
+                sizeInMb);
+
+            logger.Log(message);
+
+            using (tracer.Step(message))
+            {
+                // If extractTargetDirectory already exists, rename it so we can delete it concurrently with
+                // the unzip (along with any other junk in the folder)
+                var cleanTask = Task.Run(() => DeleteFilesAndDirsExcept(sourceZipFile, null, tracer));
```

Comment:
tbh, I just copied this from @nickwalkmsft's existing logic in the other `LocalZipFetch` method. I'm not sure why we need to explicitly clean up temp here. 

The code flow in this PR is a bit odd though. From the original design from Nick, this class `PushDeploymentController ` handles saving the zip into a temp file under `d:\local` then the Deployment manager calls a delegate defined in this class that does the deployment (unzip).

This change plugs into the same existing code where this class will save the zip under a temp folder, then the deployment manager will call `LocalZipFetchAndDeploy` which just copies it to the right place. 

I'm thinking of making this controller write the file in the right place and have the deployment manager no-op the deployment, but not sure if that impacts anything else. Mainly thinking of deployment logs that come from the deployment manager object. I'll test it to see what that does

Reply:
Right, that would seem a bit better, unless it means adding way more complexity. It's not terrible to have it in TEMP during transition as long as it gets cleaned up.